src/Lecture1.hs

```diff
@@ -94,7 +101,10 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> String -> String
+subString start end str = take (end - start' + 1) (drop start' str)
+  where
+    start' = max 0 start
```

Comment:
Nice! Maybe you could do the same for `end'`?
But generally, `drop` and `take` can work with negative input as well.